File: figma-mcp-server/find-poc-design.mjs
#!/usr/bin/env node

/**
 * Find and extract POC design from Figma
 */

import fetch from 'node-fetch';
import fs from 'fs';

const FIGMA_API_TOKEN = process.env.FIGMA_API_TOKEN;
const FILE_KEY = 'yFsZrVGKYZvCnaL7AcnU8n';

if (!FIGMA_API_TOKEN) {
  console.error('Error: FIGMA_API_TOKEN environment variable is not set');
  process.exit(1);
}

async function findPOCDesign() {
  try {
    console.log('📥 Fetching Figma file...\n');
    
    const response = await fetch(`https://api.figma.com/v1/files/${FILE_KEY}`, {
      headers: {
        'X-Figma-Token': FIGMA_API_TOKEN,
      },
    });

    if (!response.ok) {
      throw new Error(`Figma API error: ${response.status} ${response.statusText}`);
    }

    const data = await response.json();
    const designs = [];
    
    // Find all frames and components
    function searchNodes(nodes, parentPath = '') {
      if (!nodes) return;
      
      for (const node of nodes) {
        const fullPath = parentPath ? `${parentPath} > ${node.name}` : node.name;
        
        if (node.type === 'FRAME' || node.type === 'COMPONENT') {
          designs.push({
            name: node.name,
            id: node.id,
            type: node.type,
            path: fullPath,
            childCount: node.children ? node.children.length : 0,
          });
        }
        
        if (node.children && node.children.length > 0) {
          searchNodes(node.children, fullPath);
        }
      }
    }

    if (data.document && data.document.children) {
      searchNodes(data.document.children);
    }
    
    // Display designs
    console.log('🎨 Available Designs in Figma:\n');
    designs.forEach((design, idx) => {
      console.log(`${idx + 1}. ${design.name}`);
      console.log(`   Type: ${design.type} | ID: ${design.id} | Elements: ${design.childCount}`);
      console.log(`   Path: ${design.path}\n`);
    });
    
    // Find POC-related designs
    const pocDesigns = designs.filter(d => 
      d.name.toLowerCase().includes('poc') || 
      d.name.toLowerCase().includes('poc named design') ||
      d.name.toLowerCase().includes('design')
    );

    if (pocDesigns.length > 0) {
      console.log('\n🔍 POC-related designs found:\n');
      pocDesigns.forEach(design => {
        console.log(`✓ ${design.name} (${design.id})`);
      });
    }
    
  } catch (error) {
    console.error('❌ Error:', error);
    process.exit(1);
  }
}

findPOCDesign();

File: figma-mcp-server-python/find_poc_design.py
#!/usr/bin/env python3

"""
Find and extract POC design from Figma
"""

import os
import requests
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

FIGMA_API_TOKEN = os.getenv("FIGMA_API_TOKEN")
FILE_KEY = 'yFsZrVGKYZvCnaL7AcnU8n'

if not FIGMA_API_TOKEN:
    print('Error: FIGMA_API_TOKEN environment variable is not set')
    exit(1)

def search_nodes(nodes, parent_path=''):
    """Recursively search for frames and components."""
    designs = []
    if not nodes:
        return designs

    for node in nodes:
        full_path = f"{parent_path} > {node['name']}" if parent_path else node['name']

        if node['type'] in ['FRAME', 'COMPONENT']:
            designs.append({
                'name': node['name'],
                'id': node['id'],
                'type': node['type'],
                'path': full_path,
                'child_count': len(node.get('children', [])),
            })

        if node.get('children') and len(node['children']) > 0:
            designs.extend(search_nodes(node['children'], full_path))

    return designs

def find_poc_design():
    try:
        print('📥 Fetching Figma file...\n')

        response = requests.get(
            f"https://api.figma.com/v1/files/{FILE_KEY}",
            headers={"X-Figma-Token": FIGMA_API_TOKEN}
        )
        response.raise_for_status()

        data = response.json()
        designs = []

        if data.get('document') and data['document'].get('children'):
            designs = search_nodes(data['document']['children'])

        # Display designs
        print('🎨 Available Designs in Figma:\n')
        for idx, design in enumerate(designs, 1):
            print(f"{idx}. {design['name']}")
            print(f"   Type: {design['type']} | ID: {design['id']} | Elements: {design['child_count']}")
            print(f"   Path: {design['path']}\n")

        # Find POC-related designs
        poc_designs = [
            d for d in designs
            if 'poc' in d['name'].lower() or
               'poc named design' in d['name'].lower() or
               'design' in d['name'].lower()
        ]

        if poc_designs:
            print('\n🔍 POC-related designs found:\n')
            for design in poc_designs:
                print(f"✓ {design['name']} ({design['id']})")

    except requests.RequestException as e:
        print(f'❌ Error: {e}')
        exit(1)

if __name__ == "__main__":
    find_poc_design()